packages/playwright-core/src/server/dom.ts

```diff
@@ -633,7 +633,7 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
         return;
       if (element.tagName !== 'INPUT')
         throw injected.createStacklessError('Node is not an HTMLInputElement');
-      if (multiple && !(element as HTMLInputElement).multiple)
+      if (multiple && !(element as HTMLInputElement).multiple && !(element as HTMLInputElement).webkitdirectory)
```

Comment:
Let's throw a similar error when you pass multiple files into directory inputs.

Reply:
I'm throwing on the client side instead, or would you prefer both for safety?

Reply:
Yeah, both sides sounds easier to understand.